junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -82,14 +86,41 @@ public interface DisplayNameGenerator {
 	String generateDisplayNameForClass(Class<?> testClass);
 
 	/**
-	 * Generate a display name for the given {@link Nested @Nested} inner test class.
+	 * Generate a display name for the given {@link Nested @Nested} inner test
+	 * class.
 	 *
-	 * <p>If it returns {@code null}, the default display name generator will be used instead.
+	 * <p>If it returns {@code null}, the default display name generator will be
+	 * used instead.
+	 *
+	 * @param nestedClass the class to generate a name for; never {@code null}
+	 * @return the display name for the nested class; never blank
+	 * @deprecated in favor of {@link #generateDisplayNameForNestedClass(List, Class)}
+	 */
+	@SuppressWarnings("DeprecatedIsStillUsed")
+	@API(status = DEPRECATED, since = "5.12")
+	@Deprecated
+	default String generateDisplayNameForNestedClass(@SuppressWarnings("unused") Class<?> nestedClass) {
+		throw new UnsupportedOperationException(
+			"Implement generateDisplayNameForNestedClass(List<Class<?>>, Class<?>) instead");
+	}
+
+	/**
+	 * Generate a display name for the given {@link Nested @Nested} inner test
+	 * class.
+	 *
+	 * <p>If it returns {@code null}, the default display name generator will be
+	 * used instead.
```

Comment:
For both of the new methods, I think it would be helpful to users to introduce `@implNote` sections analogous to what we have for `generateDisplayNameForMethod(Class, Method)`, perhaps something along the lines of the following...

```suggestion
	 * used instead.
	 *
	 * @implNote The classes supplied as {@code enclosingInstanceTypes} may differ
	 * from the classes returned from invocations of {@link Class#getEnclosingClass()}
	 * &mdash; for example, when a nested test class is inherited from a superclass.
```

Reply:
Done in 346442e6783ffdbdadaef1764a8975d033a79fc2